fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -141,39 +145,47 @@ internal class HttpClient(
         return Response(
-            url = request.url,
-            headers = headers,
-            contentLength = contentLength ?: -1,
-            statusCode = connection.responseCode,
-            responseMessage = connection.responseMessage.orEmpty(),
-            body = DefaultBody.from(
-                {
-                    ProgressInputStream(
-                        decodeContent(
-                            stream = decodeTransfer(safeDataStream(connection), transferEncoding),
-                            encoding = contentEncoding,
-                            shouldDecode = shouldDecode
-                        ),
-                        onProgress = { readBytes ->
-                            request.responseProgress(readBytes, contentLength ?: readBytes)
-                        }
-                    ).buffered(FuelManager.progressBufferSize)
-                },
-                { contentLength ?: -1 }
-            )
+                url = request.url,
+                headers = headers,
+                contentLength = contentLength ?: -1,
+                statusCode = connection.responseCode,
+                responseMessage = connection.responseMessage.orEmpty(),
+                body = DefaultBody.from(
+                        {
+                            ProgressInputStream(
+                                    decodeContent(
+                                            stream = decodeTransfer(safeDataStream(connection, inputStream), transferEncoding),
+                                            encoding = contentEncoding,
+                                            shouldDecode = shouldDecode
+                                    ),
+                                    onProgress = { readBytes ->
+                                        request.responseProgress(readBytes, contentLength
+                                                ?: readBytes)
+                                    }
+                            ).buffered(FuelManager.progressBufferSize)
+                        },
+                        { contentLength ?: -1 }
+                )
         )
     }
 
-    private fun safeDataStream(connection: HttpURLConnection): InputStream? {
+    private fun safeDataStream(connection: HttpURLConnection, inputStream: InputStream): InputStream? {
         return try {
-            (connection.errorStream ?: connection.inputStream) ?.let { BufferedInputStream(it) }
+            BufferedInputStream((connection.errorStream ?: inputStream))
         } catch (exception: IOException) {
             // Stream error
-            try { (connection.errorStream ?: connection.inputStream).close() } catch (_: IOException) {}
+            stethoHook?.httpExchangeFailed(exception)
+            try {
+                (connection.errorStream ?: connection.inputStream).close()
+            } catch (_: IOException) {
+                stethoHook?.httpExchangeFailed(exception)
```

Comment:
I don't think you need this. This happens when closing the stream fails, not when an HttpExchange fails.